freqtrade/__init__.py

```diff
@@ -1,12 +1,11 @@
 """ Freqtrade bot """
-__version__ = 'develop'
-
-if __version__ == 'develop':
+__version__ = '2022.2.1-dev'
```

Comment:
if you think it's necessary to rename "develop" to "dev" - then maybe.
Our versions are postfixed with the git version (so develop becomes develop-11c76c3c8 - or in the renamed case, `dev-11c76c3c8`).

There's 0 benefit on having a version number there other than merge conflicts and more work (the need to do an unnecessary commit on develop after doing a release).

Poetry is not a standard installation tool in my eyes.
If it has problems with certain formats, best open an issue there.

Installation using pip (`pip install -e .`) works perfectly fine with using both develop - as well as develop-<shahash>.

Reply:
Hmm, I think I might have miscommunicated.

It's not so much that poetry is not working, it's more that freqtrade is doing it the wrong way (according to python docs linked above). Poetry just has exposed this, any other tool that is working with version numbers might balk at this as well, because it's normalised.

Any ways, I don't really care, I just got into this and fixed it as much as I could while keeping it simple. I understand that you don't want to add extra work for virtual no gain. If it happens to be needed to change, this might be a beginning.

Maybe it would be interesting to look how other version their development versions. Again, it's just the develop branch, so one could also argue it doesn't matter at all. I just think, if it doesn't matter, one could also try to follow standard notation if this is the canonical way to do it and other tools might expect it to respect it.

Reply:
If it's auto-incrementing (the way we had it) - i'm not per against it in general.
I'm however not a fan of "auto-incrementing" versions via CI (the way ccxt does it) - as this creates 1000nds of ci commits with "version bump" - which can make repositories explode quite quickly.


For the develop version - it has shown beneficial to have the commit in the version.
This allows (without any effort) to immediately identify "is this person running the latest version" or not - and if not - from when the version is - by simply using the commit in the value on either github search or `git log <commit>`.

While i'm aware of the python documentation above - it's mainly relevant if you push develop versions to pypi - which we do not (and probably will not do anytime soon).


---

your proposal of having a version `2022.2.1-dev-24f480b4c` would however also not respect the above links - as the last part (after dev) should be a number (2022.2.1-dev0`) - which will however make it impossible to determine "at a glance" which precise version we're talking about.